templates/nextflow-s3-template.yaml

```diff
@@ -95,62 +98,89 @@ Resources:
               - !Sub arn:aws:s3:::${Bucket}
               - !Sub arn:aws:s3:::${Bucket}/*
 
-          - Sid: ReadWriteAccess
-            # gives grantees read/write write access to the bucket with required --acl
-            Effect: Allow
-            Principal:
-              AWS: !Split
-                - ","
-                - !Join
-                    - ","
-                    - - Fn::ImportValue: !Sub "${AWS::Region}-nextflow-forge-service-user-ServiceUserArn"
-                      - !Join
-                          - ","
-                          - !Ref ReadWriteAccessArns
-            Action:
-              - "s3:*Object*"  
-            Condition:
-              StringEquals:
-                s3:x-amz-acl: bucket-owner-full-control
-            Resource:
-              - !Sub arn:aws:s3:::${Bucket}/*    
+          - !If
+            - NoReadWriteAccessArns
+            - Sid: ForgeUserReadWriteAccess
+              # gives forge user read/write write access to the bucket if no read/write arns are supplied
+              Effect: Allow
+              Principal:
+                AWS:
+                  - Fn::ImportValue: !Sub "${AWS::Region}-nextflow-forge-service-user-ServiceUserArn"
+              Action:
+                - "s3:*Object*"  
+              Condition:
+                StringEquals:
+                  s3:x-amz-acl: bucket-owner-full-control
+              Resource:
+                - !Sub arn:aws:s3:::${Bucket}/*   
+            - !Ref AWS::NoValue 
 
-          - Sid: ReadWriteBucketAccess
-            # gives grantees with read/write access read access to the bucket
-            Effect: Allow
-            Principal:
-              AWS: !Split
-                - ","
-                - !Join
-                    - ","
-                    - - Fn::ImportValue: !Sub "${AWS::Region}-nextflow-forge-service-user-ServiceUserArn"
-                      - !Join
-                          - ","
-                          - !Ref ReadWriteAccessArns
-            Action:
-              - "s3:AbortMultipartUpload"
-              - "s3:ListMultipartUploadParts"
-              - "s3:ListBucket*"
-              - "s3:GetBucketLocation"   
-            Resource:
-              - !Sub arn:aws:s3:::${Bucket}
-              - !Sub arn:aws:s3:::${Bucket}/*
+          - !If
+            - HasReadWriteAccessArns
+            - Sid: ReadWriteAccess
+              # gives grantees read/write write access to the bucket with required --acl
+              Effect: Allow
+              Principal:
+                AWS: !Split
+                  - ","
+                  - !Join
+                      - ","
+                      - - Fn::ImportValue: !Sub "${AWS::Region}-nextflow-forge-service-user-ServiceUserArn"
```

Comment:
Since the Tower Forge user doesn't need the `bucket-owner-full-control` condition, I would revert to granting `s3:*` to this role and only split off the `s3:*Object*` actions for the `ReadWriteAccessArns`.